proseco/core.py

```diff
@@ -156,6 +197,96 @@ def __init__(self, data=None, print_log=False, **kwargs):
         for name in ('ra', 'dec', 'RA_PMCORR', 'DEC_PMCORR'):
             self._default_formats[name] = '.6f'
 
+    def set_kwargs(self, **kwargs):
+        for name, val in kwargs.items():
+            if name in self.allowed_kwargs:
+                setattr(self, name, val)
+            else:
+                raise ValueError(f'unexpected keyword argument "{name}"')
+
+        # If an explicit obsid is not provided to all getting parameters via mica
+        # then all other params must be supplied.
+        all_pars = all(getattr(self, x) is not None for x in self.required_attrs)
+        if self.obsid == 0 and not all_pars:
+            raise ValueError('if `obsid` not supplied then all other params are required')
+
+        # If not all params supplied then get via mica for the obsid.
+        if not all_pars:
+            from mica.starcheck import get_starcheck_catalog
+            self.log(f'getting starcheck catalog for obsid {self.obsid}')
+
+            obs = get_starcheck_catalog(self.obsid)
+            obso = obs['obs']
+
+            if self.att is None:
+                self.att = [obso['point_ra'], obso['point_dec'], obso['point_roll']]
+            if self.date is None:
+                self.date = obso['mp_starcat_time']
+            if self.t_ccd is None:
+                self.t_ccd = obs.get('pred_temp', -15.0)
+            if self.man_angle is None:
+                self.man_angle = obs['manvr']['angle_deg'][0]
+            if self.detector is None:
+                self.detector = obso['sci_instr']
+            if self.sim_offset is None:
+                self.sim_offset = obso['sim_z_offset_steps']
+            if self.focus_offset is None:
+                self.focus_offset = 0
+
+            if self.n_guide is None:
+                fid_or_mon = (obs['cat']['type'] == 'FID') | (obs['cat']['type'] == 'MON')
+                self.n_guide = 8 - np.count_nonzero(fid_or_mon)
+
+            for dither_attr in ('dither_acq', 'dither_guide'):
+                if getattr(self, dither_attr) is None:
+                    dither_y_amp = obso.get('dither_y_amp')
+                    dither_z_amp = obso.get('dither_z_amp')
+                    if dither_y_amp is not None and dither_z_amp is not None:
+                        setattr(self, dither_attr, ACABox((dither_y_amp, dither_z_amp)))
```

Comment:
Somewhat in the weeds for this PR, but you could us the rules from the matlab tools to set this up to run on big dither observations as well to have working tests from mica.starcheck catalogs for those.   So, if either dither axis > 30?, set dither_acq to be square dither appropriate for the detector (20,20 for hrc etc).

Reply:
And I just checked the 30 number in characteristics and it matches.

The code I was mentioning that is going to do this from the Matlab side is 

https://github.com/sot/fot_matlab_tools/blob/master/MissionScheduling/schedule/ORs/UpdateORStarSelectors.m#L73

but that part of characteristics is not in our repo.